src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -173,44 +197,74 @@ public static bool HasUsingStatement(this SyntaxNode node, string namespaceName)
+            }
+
+            if (string.IsNullOrEmpty(namespaceName))
+            {
+                throw new ArgumentException($"'{nameof(namespaceName)}' cannot be null or empty.", nameof(namespaceName));
+            }
+
+            if (documentRoot.RootIncludesImport(namespaceName))
+            {
+                return documentRoot;
+            }
+
+            var usingDirective = CS.SyntaxFactory.UsingDirective(CS.SyntaxFactory.ParseName(namespaceName).WithLeadingTrivia(CS.SyntaxFactory.Whitespace(" ")))
+                .WithTrailingTrivia(CS.SyntaxFactory.CarriageReturnLineFeed);
+            return documentRoot.AddUsings(usingDirective);
+        }
+
+        public static VBSyntax.CompilationUnitSyntax AddImportIfMissing(this VBSyntax.CompilationUnitSyntax documentRoot, string namespaceName)
+        {
+            if (documentRoot is null)
+            {
+                throw new ArgumentNullException(nameof(documentRoot));
+            }
+
+            if (string.IsNullOrEmpty(namespaceName))
+            {
+                throw new ArgumentException($"'{nameof(namespaceName)}' cannot be null or empty.", nameof(namespaceName));
             }
 
-            return node.AncestorsAndSelf().Any(n => n.HasUsingStatement(namespaceName));
+            if (documentRoot.RootIncludesImport(namespaceName))
+            {
+                return documentRoot;
+            }
+
+            var importsStatement = VB.SyntaxFactory.ImportsStatement(VB.SyntaxFactory.SingletonSeparatedList<VBSyntax.ImportsClauseSyntax>(VB.SyntaxFactory.SimpleImportsClause(VB.SyntaxFactory.ParseName(namespaceName).WithLeadingTrivia(VB.SyntaxFactory.Whitespace(" ")))))
+                .WithTrailingTrivia(VB.SyntaxFactory.CarriageReturnLineFeed);
+            return documentRoot.AddImports(importsStatement);
         }
 
         /// <summary>
-        /// Adds a using directive for a given namespace to the document root only if the directive is not already present.
+        /// Determines if a document root element contains a using/import statement for a given namespace.
         /// </summary>
-        /// <param name="documentRoot">The document to add the directive to.</param>
-        /// <param name="namespaceName">The namespace to reference with the using directive.</param>
-        /// <returns>An updated document root with the specific using directive.</returns>
-        public static CSSyntax.CompilationUnitSyntax AddUsingIfMissing(this CSSyntax.CompilationUnitSyntax documentRoot, string namespaceName)
+        /// <param name="documentRoot">The document root to analyze.</param>
+        /// <param name="namespaceName">The namespace name to look for an import for.</param>
+        /// <returns>True if the documentRoot is a root node and contains a top-level using/import statement for the specified namespace name.</returns>
+        private static bool RootIncludesImport(this SyntaxNode documentRoot, string namespaceName)
```

Comment:
nit: this process of determining if a class includes an import seems to be its own unit of work. seeing there are a few private helper methods has me wanting a new class to isolate these helpers as part of a single unit of work.

Reply:
I refactored this a bit to try keep it more condensed. If it grows any more, I'll look into moving it into its own class.